### DISCUSSION TITLE
Is there a way to add a specific command to the command palette?

### DISCUSSION THREAD

**Discussion by rafli-sss** [wezterm/wezterm#4499] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

Mutter

### WezTerm version

20231023_015614_84ad0400

### Ask your question!

Is there a way to add specific commands to the command palette that will be directly pasted or executed to the terminal? for example :
command = "sudo systemctl start x-service"
label = "start x service"
icon = "x"

**Comment by nekowinston** [comment]
When you add entries to the [Launcher Menu](https://wezfurlong.org/wezterm/config/launch.html#the-launcher-menu), those entries are propagated to the command palette.

e.g.
```lua
local wezterm = require("wezterm")
local c = wezterm.config_builder()
c.launch_menu = {
  {
    label = "View system activity",
    args = { "btop" },
  },
}
return c
```
![2023-10-28-113833@2x](https://github.com/wez/wezterm/assets/79978224/2d512989-9e2a-491b-9c1f-b71681667050)

**Reply by rafli-sss** [reply]
Great, How do I do this if i want the command is executed in the currently active pane/tab instead of new tab?

**Reply by nekowinston** [reply]
I'd personally use [InputSelector](https://wezfurlong.org/wezterm/config/lua/keyassignment/InputSelector.html#example-of-choosing-some-canned-text-to-enter-into-the-terminal) for that purpose.

**Reply by nekowinston** [reply, marked as answer]
@rafli-sss I forgot about [augment-command-palette](https://wezfurlong.org/wezterm/config/lua/window-events/augment-command-palette.html), this should do exactly what you want.

**Reply by rafli-sss** [reply]
Yeah, thank you. Is the callback function executed every time the command palette is displayed?. Will this affect performance if I add many additional custom command entries?

**Reply by nekowinston** [reply]
From what I can see
https://github.com/wez/wezterm/blob/fec90ae04bf448d4b1475ba1d0ba1392846a70d6/wezterm-gui/src/termwindow/palette.rs#L104
https://github.com/wez/wezterm/blob/fec90ae04bf448d4b1475ba1d0ba1392846a70d6/wezterm-gui/src/termwindow/palette.rs#L236
https://github.com/wez/wezterm/blob/fec90ae04bf448d4b1475ba1d0ba1392846a70d6/wezterm-gui/src/termwindow/mod.rs#L2956
it is augmented every time the Command Palette is opened, but there might be caching that I'm not aware of.
The docs also say
> This event is emitted when the [Command Palette](https://wezfurlong.org/wezterm/config/lua/keyassignment/ActivateCommandPalette.html) is shown.
